creator-node/test/findReplicaSetUpdates.jobProcessor.test.js

```diff
@@ -0,0 +1,284 @@
+    })
+  }
+
+  async function runAndVerifyJobProcessor({
+    expectedUnhealthyReplicas = [],
+    jobProcessorArgs = {}
+  }) {
+    return expect(
+      runJobProcessor(jobProcessorArgs)
+    ).to.eventually.be.fulfilled.and.deep.equal({
+      jobsToEnqueue: expectedUnhealthyReplicas?.length
+        ? {
+            [QUEUE_NAMES.STATE_RECONCILIATION]: [
+              {
+                jobName: JOB_NAMES.UPDATE_REPLICA_SET,
+                jobData: {
+                  wallet,
+                  userId: user_id,
+                  primary,
+                  secondary1,
+                  secondary2,
+                  unhealthyReplicas: expectedUnhealthyReplicas,
+                  replicaSetNodesToUserClockStatusesMap:
+                    jobProcessorArgs?.replicaSetNodesToUserClockStatusesMap ||
+                    DEFAULT_CLOCK_STATUSES_MAP
+                }
+              }
+            ]
+          }
+        : {}
+    })
+  }
+
+  it('issues update for mismatched spIds when this node is primary', async function () {
+    // Make this node be the user's primary
+    config.set('creatorNodeEndpoint', primary)
+
+    // Verify job outputs the correct results: secondary1 should be removed from replica set because its spId mismatches
+    return runAndVerifyJobProcessor({
+      jobProcessorArgs: {
+        cNodeEndpointToSpIdMap:
+          CNODE_ENDOINT_TO_SP_ID_MAP_WHERE_SECONDARY1_MISMATCHES
+      },
+      expectedUnhealthyReplicas: [secondary1]
+    })
+  })
+
+  it('issues update for mismatched spIds when this node is secondary', async function () {
+    // Make this node be the user's secondary
+    config.set('creatorNodeEndpoint', secondary2)
+
+    // Verify job outputs the correct results: secondary1 should be removed from replica set because its spId mismatches
+    return runAndVerifyJobProcessor({
+      jobProcessorArgs: {
+        cNodeEndpointToSpIdMap:
+          CNODE_ENDOINT_TO_SP_ID_MAP_WHERE_SECONDARY1_MISMATCHES
+      },
+      expectedUnhealthyReplicas: [secondary1]
+    })
+  })
+
+  it('issues update for unhealthy primary when this node is secondary and primary fails extra health check', async function () {
+    // Make this node be the user's secondary
+    config.set('creatorNodeEndpoint', secondary1)
+
+    // Verify job outputs the correct results: primary should be removed from replica set because it's unhealthy
+    return runAndVerifyJobProcessor({
+      jobProcessorArgs: {
+        // Mark a primary as unhealthy to trigger a replica set update (if its extra health check also fails)
+        unhealthyPeers: [primary],
+        // Make the primary fail the extra health check so it gets removed from the user's replica set
+        isPrimaryHealthyInExtraHealthCheck: false
+      },
+      expectedUnhealthyReplicas: [primary]
+    })
+  })
+
+  it('does not issue update for unhealthy primary when this node is secondary and primary passes extra health check', async function () {
```

Comment:
is this bc we do multiple health checks on the primary to super make sure that the primary is unhealthy b4 rolling off?

Reply:
yeah exactly, that's been the existing behavior: do an extra health check and allow it to fail for a little while before marking it as actually unhealthy. otherwise it might've just been restarting and we'd be resyncing data too eagerly when it would've came back online a few seconds later